backend/instance_manager.go

```diff
@@ -0,0 +1,283 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+)
+
+// InstanceDisposer interface marking
+type InstanceDisposer interface {
+	Dispose()
+}
+
+// AppInstance defines the interface for application plugin instances.
+// There will be a maximum of one instance per Grafana organization.
+type AppInstance interface {
```

Comment:
Is there a way these are not (exported) interfaces? That is where we will run into breaking change by trying to add a new Handler to a plugin type such as App or Datasource.

Reply:
That's a good point :+1: One way would be to let `AppInstanceProviderFunc` return a struct instead similar to `ServeOpts`, but I'm not sure how that will work. Will try to push such an example to get a feel for it.